### PR TITLE
Add ability to update monitoring_service on google_container_cluster

### PR DESCRIPTION
**Pull request by drzero42**
My proposed fix for #308.
I'm not entirely sure how to write a test for it - I was hoping to see how updating node_version was tested, but alas there does not seem to be a test for that.

### CODE REVIEW

**Comment by drebes** [comment]
For a similar test, check either ``TestAccContainerCluster_withAdditionalZones`` or ``TestAccContainerCluster_withLegacyAbac``. Testing updates is basically having multiple steps on the test, one with the initial config, the other with the updated config, and checking that the resource changed accordingly. Using the default ``testAccCheckContainerCluster`` check from the resource should be OK as it generally just confirms that the attributes set in the resource and the parameters read from the API match, and it already tests for ``monitoring_service``.

**Comment by danawillow** [comment]
Test for updating versions: https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_container_cluster_test.go#L142

File: website/docs/r/container_cluster.html.markdown [hunk 1]
```diff
@@ -13,9 +13,20 @@ Creates a GKE cluster. For more information see
 and
 [API](https://cloud.google.com/container-engine/reference/rest/v1/projects.zones.clusters).
 
-!> **Warning:** Due to limitations of the API, all arguments except
-`node_version` are non-updateable. Changing any will cause recreation of the
-whole cluster!
+!> **Warning:** Due to limitations of the API, the following arguments are non-updateable:
```
Comment (danawillow) [inline website/docs/r/container_cluster.html.markdown]: Honestly I'm not sure this warning is particularly helpful, and we don't bother showing a similar one for most resources. Do you think it's ok to remove overall or do you think it's better to leave it in?

**Comment by danawillow** [comment]
@drzero42 are you still working on this? Otherwise I'd like to take it over.

**Comment by drzero42** [comment]
@danawillow Kind of busy with other things, so you go ahead and take it :)

**Comment by danawillow** [comment]
Closing in favor of #598.

**Pull request closed without merging**